lib/waistband/index.rb

```diff
@@ -314,6 +349,24 @@ def base_config_name
         "#{@index_name}_#{::Waistband.config.env}"
       end
 
+      def check_permission!(permission)
+        raise "::Waistband::Errors::Permissions::#{permission.classify}".constantize.new("Don't have enough permissions to #{permission}") unless check_permission?(permission)
+      end
+
+      def check_permission?(permission)
+        !!permissions[permission]
+      end
+
+      def permissions
+        @permissions ||= (config['permissions'] || {}).reverse_merge({
```

Comment:
Did you pick these defaults to be backwards-compatibile?  I would suggest more safety here for the defaults (delete = false, destroy = false).  Perhaps default_permissions should be its own method, to it can be overwritten?